dash/include/dash/matrix/internal/Matrix-inl.h

```diff
@@ -80,13 +80,54 @@ ::Matrix(
 
 template <typename T, dim_t NumDim, typename IndexT, class PatternT>
 inline Matrix<T, NumDim, IndexT, PatternT>
+::Matrix(
+  self_t && other)
+: _team(other._team),
+  _size(other._size),
+  _lcapacity(other._lcapacity),
+  _begin(other._begin),
+  _pattern(other._pattern),
+  _glob_mem(other._glob_mem),
+  _lbegin(other._lbegin),
+  _lend(other._lend),
+  _ref(other._ref)
+{
+    // do not free other globmem
+    other._glob_mem = nullptr;
```

Comment:
To be on the safe side, I'd also reset `_lbegin` and `_lend` etc in `other`, otherwise those pointer will dangle.